apps/meteor/app/livechat/imports/server/rest/visitors.ts

```diff
@@ -1,50 +1,185 @@
+import { deprecationWarning } from '../../../../api/server/helpers/deprecationWarning';
+import {
+	findVisitorInfo,
+	findVisitedPages,
+	findChatHistory,
+	searchChats,
+	findVisitorsToAutocomplete,
+	findVisitorsByEmailOrPhoneOrNameOrUsername,
+	findVisitorsToAutocompleteByName,
+} from '../../../server/api/lib/visitors';
 
 API.v1.addRoute(
-	'livechat/:rid/messages',
-	{ authRequired: true, permissionsRequired: ['view-l-room'] },
+	'livechat/visitors.info',
+	{ authRequired: true },
+	{
+		async get() {
+			check(this.queryParams, {
+				visitorId: String,
+			});
+
+			const visitor = await findVisitorInfo({ userId: this.userId, visitorId: this.queryParams.visitorId });
+
+			return API.v1.success(visitor);
+		},
+	},
+);
+
+API.v1.addRoute(
+	'livechat/visitors.pagesVisited/:roomId',
+	{ authRequired: true },
 	{
 		async get() {
 			check(this.urlParams, {
-				rid: String,
+				roomId: String,
+			});
+			const { offset, count } = this.getPaginationItems();
+			const { sort } = this.parseJsonQuery();
+
+			const pages = await findVisitedPages({
+				userId: this.userId,
+				roomId: this.urlParams.roomId,
+				pagination: {
+					offset,
+					count,
+					sort,
+				},
 			});
 
+			return API.v1.success(pages);
+		},
+	},
+);
+
+API.v1.addRoute(
+	'livechat/visitors.chatHistory/room/:roomId/visitor/:visitorId',
+	{ authRequired: true },
+	{
+		async get() {
+			check(this.urlParams, {
+				visitorId: String,
+				roomId: String,
+			});
 			const { offset, count } = this.getPaginationItems();
 			const { sort } = this.parseJsonQuery();
+			const history = await findChatHistory({
+				userId: this.userId,
+				roomId: this.urlParams.roomId,
+				visitorId: this.urlParams.visitorId,
+				pagination: {
+					offset,
+					count,
+					sort,
+				},
+			});
 
-			const room = LivechatRooms.findOneById(this.urlParams.rid);
+			return API.v1.success(history);
+		},
+	},
+);
 
-			if (!room) {
-				throw new Error('invalid-room');
-			}
+API.v1.addRoute(
+	'livechat/visitors.searchChats/room/:roomId/visitor/:visitorId',
+	{ authRequired: true },
+	{
+		async get() {
+			check(this.urlParams, {
+				visitorId: String,
+				roomId: String,
+			});
+			const { roomId, visitorId } = this.urlParams;
+			const { searchText, closedChatsOnly, servedChatsOnly } = this.queryParams;
+			const { offset, count } = this.getPaginationItems();
+			const { sort } = this.parseJsonQuery();
+			const history = await searchChats({
+				userId: this.userId,
+				roomId,
+				visitorId,
+				searchText,
+				closedChatsOnly: closedChatsOnly === 'true',
+				servedChatsOnly: servedChatsOnly === 'true',
+				pagination: {
+					offset,
+					count,
+					sort,
+				},
+			});
+			return API.v1.success(history);
+		},
+	},
+);
 
-			if (!canAccessRoom(room, this.user)) {
-				throw new Error('not-allowed');
+/* deprecated endpoint - "livechat/visitors.autocomplete", use "livechat/visitors.autocompleteByName" instead */
+API.v1.addRoute(
```

Comment:
Can you move this deprecation to its own PR? 🙈

Reply:
Alright Done!! :grimacing: